src/code_1b40_len_20b0.c

```diff
@@ -34,9 +35,155 @@ Gfx D_80074230[] = {
     gsSPEndDisplayList(),
 };
 
-INCLUDE_ASM(void, "code_1b40_len_20b0", step_game_loop);
+extern DisplayContext D_80164000[];
+extern s16 D_8009A690;
 
-INCLUDE_ASM(s32, "code_1b40_len_20b0", gfx_task_background);
+void gfx_init_state(void);
+void gfx_draw_background(void);
+
+void step_game_loop(void) {
+    PlayerData* playerData = &gPlayerData;
+    s8* temp80074021;
+    s16* temp80074182;
+    s16* temp800741A0;
+    s32* overrideFlags;
+    s16* temp8009A690;
+
+    update_input();
+
+    (*gGameStatusPtr)->frameCounter++;
+
+    playerData->frameCounter += 2;
+    if (playerData->frameCounter > 215999999) {
+        playerData->frameCounter = 215999999;
+    }
+
+    update_max_rumble_duration();
+
+    temp80074021 = &D_80074021;
+    if (*temp80074021 != 0) {
+        (*temp80074021)-- ;
+        if (*temp80074021 == 0) {
+            *temp80074021 = D_80074020;
+        } else {
+            return;
+        }
+    }
+
+    func_8011BAE8();
+    func_8003857C();
+    update_dynamic_entities();
+    update_triggers();
+    update_scripts();
+    update_messages();
+    update_menu_icons();
+    step_current_game_mode();
+    update_entities();
+    func_80138198();
+    func_8014A548();
+    update_ambient_sounds();
+    func_80149734();
+    update_windows();
+    update_curtains();
+
+    overrideFlags = &gOverrideFlags;
```

Comment:
Given that you're using anonymous blocks in subsequent sections this should probably be in one too